### DISCUSSION TITLE
global.features settings?

### DISCUSSION THREAD

**Discussion by saper** [DOCGroup/ACE_TAO#2167] [Q&A] [answered]
I could not find any decent description of the features that can/should be set to build ACE. 

What are the typical settings used on a modern Linux/BSD system? I just want to learn/study the platform.

Random questions: What do things like `inet_ssl` do? Which `xerces` should I set?

ACE_INSTALL.html refers to some features as "Makefile flags" but I think this is all now done with MPC?

Is `use_wchar` used and working? ([not quite for me](https://github.com/DOCGroup/ACE_TAO/issues/2145#issuecomment-1826190604))

**Comment by mitza-oci** [comment, marked as answer]
> What are the typical settings used on a modern Linux/BSD system? I just want to learn/study the platform.

I suggest starting with the GitHub Actions configurations that are available in the repository such as `.github/workflows/linux.yml`

There's also the distributed scoreboard at https://www.dre.vanderbilt.edu/scoreboard/integrated.html where each build has a `[Config]` link that shows its configuration.

> Random questions: What do things like `inet_ssl` do? 

For `inet_ssl` specifically, it may not do anything at all.

> Which `xerces` should I set?

An example of using xerces3 is in the linux.yml file referenced above.  Other xerces versions are too old to use.

> ACE_INSTALL.html refers to some features as "Makefile flags" but I think this is all now done with MPC?

Both MPC features and makefile flags are relevant to the build (on platforms that use GNU Make).

> Is `use_wchar` used and working? ([not quite for me](https://github.com/DOCGroup/ACE_TAO/issues/2145#issuecomment-1826190604))

Probably more tested on Win32 where uses_wchar itself (meaning accessing an OS-level API using wchar_t as the predominant character type) is more used.
